### PR TITLE
kexec: restore static ips after kexec

### PR DESCRIPTION
**Pull request by Lassulus**
Initially I wanted to use jq, but for the sake of maintainability and readability I sided with python :)

### CODE REVIEW

File: nix/kexec-installer.nix [hunk 1]
```diff
@@ -29,6 +33,16 @@
     for p in /etc/ssh/ssh_host_*; do
       cp -a "$p" ssh
     done
+
+    # save the networking config for later use
+    if ! type -p ip &>/dev/null; then
```
Comment (Lassulus) [inline nix/kexec-installer.nix:38]: I guess the `!` is wrong here, will push fix

File: nix/kexec-installer.nix [hunk 2]
```diff
@@ -95,7 +95,7 @@ in {
       pkgs.iproute2
     ];
     wantedBy = [ "multi-user.target" ];
-    after = [ "network.target" ];
+    after = [ "network-online.target" ];
```
Comment (Mic92) [inline nix/kexec-installer.nix:98]: Could this leave the network without static ip addresses when the network-online.target fails i.e. there is no dhcp?

Comment (Lassulus) [reply nix/kexec-installer.nix:98]: maybe. but it worked on the testinstance

**Comment by Mic92** [comment]
I switched to generating networkd units. This should also help with interfaces not being visible at boot immediately.
I also added v4/v6 support.

**Comment by Lassulus** [comment]
I will test this again. last time I got nonsense fe80 addresses on the interface which set the default route to nowhere

**Comment by Mic92** [comment]
> I will test this again. last time I got nonsense fe80 addresses on the interface which set the default route to nowhere

I filtered those out by checking for the scope of the ip.

**Comment by Mic92** [comment]
The test now checks dhcpv4, router advertisement and static ip addresses + routes.
I think this is good to go.

**Pull request merged**